pydantic/dataclasses.py

```diff
@@ -232,6 +233,13 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         else:
             frozen_ = config_wrapper.frozen or False
 
+        # It's not recommended to define both, as the setting from the dataclass decorator will take priority.
+        if frozen is not None and config_wrapper.frozen == True:
+            warn(
+                "Both 'dataclass' decorator and 'config.frozen' are defined. 'config.frozen' take priority.",
+                UserWarning,
+            )
```

Comment:
```suggestion
if frozen is not None:
    frozen_ = frozen
    if config_wrapper.frozen:
        # It's not recommended to define both, as the setting from the dataclass decorator will take priority.
        warn(...)
else:
    frozen_ = config_wrapper.frozen or False
```

Maybe?

Reply:
Thank you for your comments, and I agree with you! 
I have fixed it